src/main/scala/akka/contrib/datareplication/ORMultiMap.scala

```diff
@@ -0,0 +1,166 @@
+/**
+ * Copyright (C) 2009-2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.contrib.datareplication
+
+import akka.cluster.{UniqueAddress, Cluster}
+
+object ORMultiMap {
+
+  /**
+   * Provides an empty multimap.
+   */
+  val empty: ORMultiMap =
+    new ORMultiMap(ORMap.empty)
+  def apply(): ORMultiMap = empty
+
+  /**
+   * Java API
+   */
+  def create(): ORMultiMap = empty
+
+  def unapply(value: Any): Option[Map[String, Set[Any]]] = value match {
+    case r: ORMultiMap ⇒ Some(r.entries)
+    case _             ⇒ None
+  }
+}
+
+/**
+ * An immutable multi-map implementation. This class wraps an
+ * [[ORMap]] with an [[ORSet]] for the map's value.
+ */
+case class ORMultiMap private (private[akka] val map: ORMap)
+    extends ReplicatedData with RemovedNodePruning with Serializable {
```

Comment:
will you add protobuf serialization in this PR, or do you want me to do that afterwards?